src/libultra_code/contpfs.c

```diff
@@ -0,0 +1,318 @@
+#include <ultra64.h>
+#include <global.h>
+
+extern __OSInode __osPfsInodeCache; // bss
+s32 	  __osPfsInodeCacheChannel = -1;
+u8		  __osPfsInodeCacheBank = 250;
+
+u16 __osSumcalc(u8* ptr, s32 length) {
+    s32 i;
+    u32 sum = 0;
+    u8* temp;
+
+    temp = ptr;
+    for (i = 0; i < length; i++) {
+        sum += *temp++;
+    }
+    return sum & 0xFFFF;
+}
+
+s32 __osIdCheckSum(u16* ptr, u16* checkSum, u16* idSum) {
+
+    u16 data = 0;
+    u32 i;
+
+    *checkSum = *idSum = 0;
+    for (i = 0; i < ((sizeof(__OSPackId) - sizeof(u32)) / sizeof(u8)); i += 2) {
+        data = *((u16*)((u32)ptr + i)); // Cast to u32 required to match
```

Comment:
id say cast comments arent really needed, its something to note for a review but it doesnt really add much to the code. imo take this and the other ones out